src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -144,8 +147,22 @@ def _check_value(self, action, value):
         # Override to customize the error message when a argument is not among the available choices
         # converted value must be one of the choices (if specified)
         if action.choices is not None and value not in action.choices:
-            msg = 'invalid choice: {}'.format(value)
-            raise argparse.ArgumentError(action, msg)
+            error_msg = "{prog}: '{value}' is not an {prog} command. See '{prog} --help'.".format(prog=self.prog,
+                                                                                                  value=value)
+            telemetry.set_user_fault(error_msg)
+            logger.error(error_msg)
+            candidates = difflib.get_close_matches(value, action.choices, cutoff=0.8)
+            if candidates:
+                print_args = {
+                    's': 's' if len(candidates) > 1 else '',
+                    'verb': 'are' if len(candidates) > 1 else 'is',
+                    'value': value
+                }
+                suggestion_msg = "\nThe most similar command{s} to '{value}' {verb}:\n".format(**print_args)
+                suggestion_msg += '\n'.join(['\t' + candidate for candidate in candidates])
+                print(suggestion_msg, file=sys.stderr)
+
+            self.exit(2)
```

Comment:
Is this the same exit code argparse would normally return?

Reply:
Yes, argparse would normally return [exit code 2](https://github.com/chewong/azure-cli/blob/bc1347bc6554a8e6fdcfed609a0e7bac1529436a/src/azure-cli-core/azure/cli/core/parser.py#L124).